design-system/react/src/utils/system.tsx

```diff
@@ -38,11 +25,16 @@ export type CreateComponent<P, OP = any> = ForwardRefExoticComponent<
  */
 
 export function createComponent<
-  P,
-  OP = unknown,
-  Props = PropsWithAs<OP extends string ? Omit<P, OP> : P>
->(render: Render<P, OP>) {
-  return forwardRef<any, Props>((props, ref) =>
-    render({ ref, ...props } as any)
+  Props,
+  CustomObjProps = unknown,
+  PropsToOmit = unknown,
+  HTMLElement = any,
+  FinalProps = PropsWithAs<
```

Comment:
This prop is intended to be received? FinalProps? If not, could use direct it on the place needed. But it is not a big deal.

Reply:
This pattern is used to create types that will be used internally in the function scope using received parameters to build them. It's often used on some projects like XState.